Main.py

```diff
@@ -362,7 +363,12 @@ def precollect_hint(location):
                         if location.name in world.start_location_hints[location.player]:
                             precollect_hint(location)
                         elif location.item.name in world.start_hints[location.item.player]:
-                            precollect_hint(location)
+                            received_item = location.item
+                            if collected_hints[received_item.player].setdefault(received_item.name, 0)\
```

Comment:
this doesn't collect a random subset of them, but rather a randomly deterministic subset that is not seeded.

Reply:
Honestly didn't consider that they should be randomized but obviously. Done.